sconstruct

```diff
@@ -545,3 +545,65 @@ env.Alias('appx',[installed_appx_storeSubmission,installed_appx_sideLoadable])
 env.Default(dist)
 
 env.SConscript("tests/sconscript", exports=["env", "sourceDir", "pot"])
+
+
+def generate_module_list(target, source, env):
+	""" 
+	Generate a list of Python modules from compiled '.pyc' files within a specified zipfile.
+
+    This function lists all '.pyc' files contained in a given zipfile, converts their paths to Python module
+    format, performs a sanity check to ensure a specific module is present, and then writes the list of
+    modules to a specified output file for packaging as an AppVeyor artefact.
+
+    Args:
+        target (list of SCons.Node.FS.File): A single element list containing the target file node where the 
+											 list of modules will be written.
+        source (list of SCons.Node.FS.File): A single element list containing the source file node to be 
+											 processed. It should contain only the zipfile from which the 
+											 '.pyc' files will be listed.
+        env (SCons.Environment): The SCons environment context under which this function is executed. This
+                                 parameter provides access to SCons construction variables, methods, and tools.
+
+    Returns:
+        None: The function does not return anything but writes the list of modules to the target file.
+
+    Raises:
+        ValueError: If the specified sanity check module (e.g., 'NVDAObjects.UIA') is not found in the
+					list of modules extracted from the zipfile.
+        FileNotFoundError: If the specified zipfile in `source` does not exist.
+	"""
+	import zipfile

```

Comment:
These are standard library modules, so they can be safely imported at the module level. This is especially noticeable for `zipfile`, which will be imported lazily multiple times.